### ISSUE TITLE
realtime monitor - cannot connect to port

### ISSUE THREAD

**Issue by kevincameronmtl** [BehaviorTree/Groot#64]
Hello,
I am trying to get the real-time monitor to work. i saw a similar issue that suggests a zmq publisher.
PublisherZMQ publisher_zmq(tree);

do you have a complete example of this.  i have tried adding into my init code, but i get the same issue.
thank you

**Comment by kevincameronmtl** [comment]
I have made some progress here. i believe i have the publisher part working. but when i try to connect using Groot i get this error message:

 deque::_M_range_check: __n (which is 755)>= this->size() (which is 27)

any idea?

**Comment by facontidavide** [comment]
No... Sorry, overwhelmed with other projects. I will try to get back to you... Eventually

**Comment by tgreier** [comment]
Hello. I am experiencing the same issue. @kevincameronmtl Did you every make progress on this?

**Comment by esromneb** [comment]
@tgreier are you trying to implement a custom zmq connection to Groot?

**Comment by ramilmsh** [comment]
@tgreier  If I were you I'd make sure to compile the latest versions of BehaviorTree.CPP Groot and your project as a single meta-project on one machine.

for me simply adding BT::PublisherZMQ publisher_zmq( tree );
worked like a charm, so maybe you have a version mismatch or something?

If that doesn't work, make a sample project with a minimal tree functionality and ZMQ publisher, which results in an error for you. Maybe I can take a look and see if it works for me

**Comment by tgreier** [comment]
My problem was caused by my BehaviorTreeEngine, where the manifest was not being copied from the factory to the Tree. Therefore there was no manifest data to load.
